Content.Server/Doors/Systems/AirlockSystem.cs

```diff
@@ -109,5 +113,60 @@ private void OnDoorPry(EntityUid uid, AirlockComponent component, BeforeDoorPryE
                 args.Cancel();
             }
         }
+
+        private void OnRangedInteract(EntityUid uid, AirlockComponent component, RangedInteractEvent args)
+        {
+            args.Handled = true;
+            var interactionSystem = EntityManager.EntitySysManager.GetEntitySystem<SharedInteractionSystem>();
+            if (!interactionSystem.InRangeUnobstructed(args.UserUid, args.TargetUid, -1f, CollisionGroup.Opaque))
+            {
+                args.Handled = false;
+                return;
+            }
+            // If it isn't a door remote we don't use it
+            if (!EntityManager.TryGetComponent<DoorRemoteComponent?>(args.UsedUid, out var remoteComponent))
+            {
+                args.Handled = false;
+                return;
+            }
+            // Remotes don't work on doors without access
+            if (!EntityManager.HasComponent<AccessReaderComponent>(component.Owner))
+            {
+                args.Handled = false;
+                return;
+            }
+            var doorComponent = EntityManager.GetComponent<ServerDoorComponent>(component.Owner);
+
+            
+            if (remoteComponent.Mode == DoorRemoteComponent.OperatingMode.OpenClose)
+            {
+                if (doorComponent.State == SharedDoorComponent.DoorState.Open)
+                {
+                    doorComponent.TryClose(args.UsedUid);
+                }
+                else if (doorComponent.State == SharedDoorComponent.DoorState.Closed)
+                {
+                    doorComponent.TryOpen(args.UsedUid);
+                }
```

Comment:
With the new door system this can just be condensed to `SharedDoorSystem.TryToggleDoor(uid)`